src/Container.php

```diff
@@ -161,50 +162,38 @@ private function resolveParams(string $qn, ReflectionMethod $constructor): itera
                 yield $param->getDefaultValue();
                 continue;
             }
+
             $paramType = $param->getType();
+
             if ($paramType instanceof ReflectionUnionType) {
```

Comment:
Instead of this if and `throwNonResolvableUnionParameterTypeException` method inside, we can create something like `$this->checkForUnionType()` that will do both